nevergrad/functions/powersystems/core.py

```diff
@@ -125,28 +121,16 @@ def get_num_vars(self) -> tp.List[tp.Any]:
     def _simulate_power_system(self, *arrays: np.ndarray) -> float:
         failure_cost = self.failure_cost  # Cost of power demand which is not satisfied (equivalent to a expensive infinite thermal group).
         dam_agents = self.dam_agents
-        # Deep, robust flattening of any nesting of iterables.
+        for agent, array in zip(dam_agents, arrays):
+            agent.set_parameters(array)
```

Comment:
That is massively simpler.
Absolutely no need to convert to a flat list to reconvert it to a list of lists which are themselves converted to arrays